packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1546,7 +1546,7 @@ class RenderFittedBox extends RenderProxyBox {
     if (_transform != null)
       return;
 
-    if (child == null) {
+    if (child == null || size.isEmpty) {
       _hasVisualOverflow = false;
```

Comment:
That's true.  Maybe we should skip this part of the change and only to the part in the `paint` function.